### PR TITLE
[nightly] Update the many nodes actor workload to 1k nodes.

### PR DESCRIPTION
**Pull request by fishbone**
Signed-off-by: Yi Cheng <74173148+iycheng@users.noreply.github.com>

## Why are these changes needed?
This PR update the test to run with 1k nodes 20k actors. Right now the infra can't scale to 1k nodes when starting because of rate limiting. This PR scale up inside the testing script slowly to avoid the rate limiting.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by scv119 — APPROVED** [review]
should we create a new test? otherwise the new test result is not comparable with the old, and we might not catch regression for 2.3 release.

**Comment by iycheng** [comment]
SG. I'll run 500/1000/1500/2000 node tests when scale things up. I think 250 test is no longer useful.

**Comment by iycheng** [comment]
the test cost too much money. i might change the instance type to tiny.

File: release/nightly_tests/many_nodes_tests/compute_config.yaml [hunk 1]
```diff
@@ -9,6 +9,6 @@ head_node_type:
 worker_node_types:
    - name: worker_node
      instance_type: m5.xlarge
-     min_workers: 250
-     max_workers: 250
+     min_workers: 500
+     max_workers: 2000
```
Comment (rkooo567) [inline release/nightly_tests/many_nodes_tests/compute_config.yaml:13]: Isn't it supposed to be 1000?

**Review by rkooo567 — COMMENTED** [review]
Isn't slow scaling should be transparently handled by the prod? Are they planning to fix that?

File: release/nightly_tests/many_nodes_tests/compute_config.yaml (hunk 1 above)
Comment (iycheng) [reply release/nightly_tests/many_nodes_tests/compute_config.yaml:13]: yes, good catch!

**Comment by iycheng** [comment]
> Isn't slow scaling should be transparently handled by the prod? Are they planning to fix that?

not yet :( I don't think it'll be fixed in short term.

File: release/nightly_tests/many_nodes_tests/compute_config.yaml (hunk 1 above)
Comment (iycheng) [reply release/nightly_tests/many_nodes_tests/compute_config.yaml:13]: i'll run 1k test and then 2k test.

**Comment by iycheng** [comment]
passed: https://buildkite.com/ray-project/release-tests-pr/builds/26633#0185f12a-6c3b-4e37-ba74-d66748b42671

**Comment by iycheng** [comment]
Hmmm, make one mistake here :( 
I'll create another one to fix it.

**Pull request merged**